### PR TITLE
Add missing catches to handlePromise

### PR DESCRIPTION
**Pull request by glekner**
This will solve Uncaught errors from `handlePromise`

### CODE REVIEW

**Comment by glekner** [comment]
@suomiy @yaacov @rawagner can you review?

**Comment by yaacov** [comment]
lgtm :+1:

I think it's better then not catching the errors :cake:

**Comment by rawagner** [comment]
we could also change the `handlePromise` implementation to something like
```javascript
const handlePromise = (promise, okCallback, errorCallback) => {
    setInProgress(true);
    promise.then(
      (res) => {
        setInProgress(false);
        setErrorMessage('');
        okCallback && okCallback(res);
      },
      (error) => {
        const errorMsg = error.message || 'An error occurred. Please try again.';
        setInProgress(false);
        setErrorMessage(errorMsg);
        errorCallback && errorCallback(errorMsg);
      },
    );
  };
```
which would not force us to `catch`
With the current implementation we do something like
```javascript
handlePromise(k8sCreate(PersistentVolumeClaimModel, pvcCloneObj)).then(close).catch(consoleError)
```
With proposed one
```javascript
handlePromise(k8sCreate(PersistentVolumeClaimModel, pvcCloneObj, close))
//or optional error callback
handlePromise(k8sCreate(PersistentVolumeClaimModel, pvcCloneObj, close, optionalErrorCallback))
```

**Comment by glekner** [comment]
@rawagner 
I changed `handlePromise` to the following

`const handlePromise = (promise, onFulfill, onError)`

if no error handler is supplied, we put it in console
`(error) => onError ? onError(errorMsg) : consoleError(errorMsg);`

File: frontend/packages/console-shared/src/utils/utils.ts [hunk 1]
```diff
@@ -43,3 +43,6 @@ export const isValidUrl = (url: string): boolean => {
     return false;
   }
 };
+
+// eslint-disable-next-line no-console
+export const consoleError = (err: any) => console.error(err);
```
Comment (suomiy) [inline frontend/packages/console-shared/src/utils/utils.ts:48]: do we want to expose this function?

File: frontend/public/components/utils/promise-component.tsx [hunk 2]
```diff
@@ -1,22 +1,23 @@
...
         setErrorMessage('');
-        return res;
+        onFulfill && onFulfill(res);
       },
       (error) => {
         const errorMsg = error.message || 'An error occurred. Please try again.';
         setInProgress(false);
         setErrorMessage(errorMsg);
-        return Promise.reject(errorMsg);
+        onError ? onError(errorMsg) : consoleError(errorMsg);
```
Comment (suomiy) [inline frontend/public/components/utils/promise-component.tsx:20]: maybe we could display the original error here?

**Comment by glekner** [comment]
fixed @suomiy suggestions

File: frontend/public/components/utils/promise-component.tsx [hunk 3]
```diff
@@ -4,19 +4,20 @@ export const withHandlePromise: WithHandlePromise = (Component) => (props) => {
...
-        return res;
+        onFulfill && onFulfill(res);
       },
       (error) => {
         const errorMsg = error.message || 'An error occurred. Please try again.';
         setInProgress(false);
         setErrorMessage(errorMsg);
-        return Promise.reject(errorMsg);
+        // eslint-disable-next-line no-console
+        onError ? onError(error) : console.error(error);
```
Comment (suomiy) [inline frontend/public/components/utils/promise-component.tsx:20]: `onError` callback should be backwards compatible. I just mean changing the console.error

**Comment by glekner** [comment]
fixed tests @rawagner @spadgett

**Comment by rawagner** [comment]
I like it, @spadgett thoughts ?

**Comment by spadgett** [comment]
/hold

Sorry needs rebase again :(

**Comment by glekner** [comment]
rebased @spadgett

**Comment by spadgett** [comment]
/hold

Build error:

```
ERROR in /go/src/github.com/openshift/console/frontend/packages/console-app/src/components/modals/restore-pvc/restore-pvc-modal.tsx
(112,91): Property 'then' does not exist on type 'void'.
```

**Comment by glekner** [comment]
@spadgett fixed

**Pull request merged**